profile/profile.go

```diff
@@ -565,7 +566,7 @@ func (p *Profile) ScaleN(ratios []float64) error {
 // symbolized function information.
 func (p *Profile) HasFunctions() bool {
 	for _, l := range p.Location {
-		if l.Mapping != nil && !l.Mapping.HasFunctions {
+		if l.Mapping != nil && !l.Mapping.Unsymbolizable() && !l.Mapping.HasFunctions {
```

Comment:
Not sure that these checks belong in a function that is called HasFunctions()
Perhaps let the symbolizer call Unsymbolizable directly?